src/applications/model/crazytaxi-streaming-server.cc

```diff
@@ -29,95 +29,102 @@
-    Ptr<ConstantRandomVariable> iat_cursor = CreateObjectWithAttributes<ConstantRandomVariable> ("Constant", DoubleValue(50));
-    AddNewTraffic (pkt_cursor, iat_cursor);
-
-    /** Add VBR audio stream */
-    // Packet size
-    Ptr<ConstantRandomVariable> pkt_vbr_audio1 = CreateObjectWithAttributes<ConstantRandomVariable> ("Constant", DoubleValue(244));
-    Ptr<ConstantRandomVariable> pkt_vbr_audio2 = CreateObjectWithAttributes<ConstantRandomVariable> ("Constant", DoubleValue(1384));
-
-    Ptr<GamingTrafficModel> pkt_vbr_audio = CreateObject<GamingTrafficModel> ();
-    pkt_vbr_audio->SetRngs ({pkt_vbr_audio1, pkt_vbr_audio2});
-    pkt_vbr_audio->SetProb ({0.0776, 0.9224});
-
-    // Packet inter-arrival time
-    Ptr<ConstantRandomVariable> iat_vbr_audio = CreateObjectWithAttributes<ConstantRandomVariable> ("Constant", DoubleValue(50));
-    AddNewTraffic (pkt_vbr_audio, iat_vbr_audio);
-
-    /** Add video stream */
-    // Packet size
-    Ptr<UniformRandomVariable> pkt_video1 = CreateObjectWithAttributes<UniformRandomVariable> ("Min", DoubleValue(0),
-                                                                                              "Max", DoubleValue(1355));
-    Ptr<ConstantRandomVariable> pkt_video2 = CreateObjectWithAttributes<ConstantRandomVariable> ("Constant", DoubleValue(1356));
-
-    Ptr<GamingTrafficModel> pkt_video = CreateObject<GamingTrafficModel> ();
-    pkt_video->SetRngs ({pkt_video1, pkt_video2});
-    pkt_video->SetProb ({0.3606, 0.6394});
-
-    // Packet inter-arrival time
-    Ptr<ConstantRandomVariable> iat_video1 = CreateObjectWithAttributes<ConstantRandomVariable> ("Constant", DoubleValue(0));
-    Ptr<LogNormalRandomVariable> iat_video2 = CreateObjectWithAttributes<LogNormalRandomVariable> ("Mu", DoubleValue(0.34),
-                                                                                                  "Sigma", DoubleValue(1.73));
-
-    Ptr<GamingTrafficModel> iat_video = CreateObject<GamingTrafficModel> ();
-    iat_video->SetRngs ({iat_video1, iat_video2});
-    iat_video->SetProb ({0.5725, 0.4275});
-
-    AddNewTraffic (pkt_video, iat_video);
-  }
-
-  void
-  CrazyTaxiStreamingServer::AddNewTraffic (Ptr<RandomVariableStream> pkt, Ptr<RandomVariableStream> iat)
-  {
-    NS_LOG_FUNCTION (this);
-    m_pksList.push_back (pkt);
-    m_iatList.push_back (iat);
-  }
-
-  void
-  CrazyTaxiStreamingServer::Send (void)
-  {
-    NS_LOG_FUNCTION (this);
-    for (unsigned i = 0; i < m_pksList.size(); ++i)
+NS_LOG_COMPONENT_DEFINE ("CrazyTaxiStreamingServer");
+
+NS_OBJECT_ENSURE_REGISTERED (CrazyTaxiStreamingServer);
+
+TypeId
+CrazyTaxiStreamingServer::GetTypeId (void)
+{
+  static TypeId tid = TypeId ("ns3::CrazyTaxiStreamingServer")
+      .SetParent<GamingStreamingServer> ()
+      .SetGroupName("Applications")
+      .AddConstructor<CrazyTaxiStreamingServer> ()
+  ;
+  return tid;
+}
+
+CrazyTaxiStreamingServer::CrazyTaxiStreamingServer ()
+  : GamingStreamingServer ()
```

Comment:
my bad, the default base class constructor is always called, no need to do it explicitly.
In you need to call a different base constructor, instead, you should call it explicitly in the initialization list.
See [here](https://www.learncpp.com/cpp-tutorial/114-constructors-and-initialization-of-derived-classes/)